src_c/display.c

```diff
@@ -1079,6 +1117,31 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
                 state->scaled_gl_w = w;
                 state->scaled_gl_h = h;
             }
+
+            /* Even if this succeeds, we can never *really* know if vsync actually works.
+               There may be screen tearing, blocking double buffering, triple buffering,
+               render-offloading where the driver for the on-board graphics *doesn't* have
+               vsync enabled, or cases where the driver lies to us because the user has
+               configured vsync to be aways on or always off, or vsync is on by default
+               for the whole desktop because of wayland GL compositing. */
+            if (vsync) {
+                if (SDL_GL_SetSwapInterval(-1) != 0){
+                    if (PyErr_WarnEx(PyExc_Warning, "adaptive vsync for OpenGL not available, trying regular", 1) != 0) {
+                        _display_state_cleanup(state);
+                        goto DESTROY_WINDOW;
+                    }
+                    if (SDL_GL_SetSwapInterval(1) != 0){
+                        if(PyErr_WarnEx(PyExc_Warning, "regular vsync for OpenGL *also* not available", 1) != 0) {
+                            _display_state_cleanup(state);
+                            goto DESTROY_WINDOW;
+                    }
```

Comment:
```suggestion
                            }
```

(Avoid confusion on about where the warning scrope ends)